### ISSUE TITLE
Enhancement: Support prerelease version numbers for PowerShell Modules

### ISSUE THREAD

**Issue by deadlydog** [rfennell/AzurePipelines#547]
I'm using v2.151.2 of the `Version PowerShell Module` task in Azure DevOps (VSTS). It works great for versioning stable PowerShell module's, but doesn't support prerelease versioning. If we try to just specify a version number like `1.0.0-alpha` then we get an error message back that the version number format is not correct. This is because in the PowerShell module manifest (.psd1) file, we can't simply put:

```powershell
ModuleVersion = '1.0.0-alpha'
```

Instead, as discussed [in this post](https://devblogs.microsoft.com/powershell/prerelease-versioning-added-to-powershellget-and-powershell-gallery/), we need to leave the `ModuleVersion` as `1.0.0`, and add a `Prerelease` variable to the  `PSData` section of the manifest file, like this:

```powershell
   ModuleVersion = '1.0.0'
   ...
      PrivateData = @{
         PSData = @{
            Prerelease = '-alpha'
      }
   }
```

It would be amazing if this task supported parsing the prerelease part of the version from the Version Number string (e.g. 1.0.0-alpha) when it exists, and updating the PowerShell module manifest file accordingly.

Thanks for the great extension!

**Comment by ChrisLGardner** [comment]
I'd been thinking about adding this at some point as we've been dealing with this a bit too. I'll see if I can find some time over the next week or so to add the functionality.

**Comment by rfennell** [comment]
I have you tried checking the "Inject Version" option on the advanced menu, this should bypass all validation on the value being passed into the task

**Comment by deadlydog** [comment]
Hey @rfennell, as described in the issue description, the problem isn't that this extension doesn't inject the version number into the file, it's that it injects the `prerelease` portion of the version number into the wrong place. For some reason the PowerShell team decided to be different and not just allow you to specify the entire version number altogether like most other apps. i.e. You cannot just do `ModuleVersion = '1.0.0-alpha'`; you have to inject the `-alpha` portion separately in another spot in the file. If you try and use `ModuleVersion = '1.0.0-alpha'` then the `Publish-Module` cmdlet will complain.

**Comment by ChrisLGardner** [comment]
Fixed in 2.19.x
